src/inet/TCPEndPointImplLwIP.cpp

```diff
@@ -856,8 +805,15 @@ err_t TCPEndPointImplLwIP::LwIPHandleIncomingConnection(void * arg, struct tcp_p
         if (err == CHIP_NO_ERROR)
         {
             TCPEndPoint * connectEndPoint = nullptr;
-            err                           = listenEP->GetEndPointManager().NewEndPoint(&connectEndPoint);
-            conEP                         = static_cast<TCPEndPointImplLwIP *>(connectEndPoint);
+#if CHIP_SYSTEM_CONFIG_NO_LOCKING
```

Comment:
So when CHIP_SYSTEM_CONFIG_NO_LOCKING this will create code that just has a thread race?

How about we:

```
#if CHIP_SYSTEM_CONFIG_NO_LOCKING
#error CHIP_SYSTEM_CONFIG_NO_LOCKING not supported along with using LwIP for TCP, because handling incoming connection attempts needs to be done synchronously in LwIP, but part of the work for it needs to happen in the Matter context.  If support for this configuration is needed, this part needs to be figured out.
#endif
```

Reply:
Yes, added a static_assert in this file which checks !CHIP_SYSTEM_CONFIG_NO_LOCKING.